src/lib/y2storage/encryption_processes/apqn.rb

```diff
@@ -0,0 +1,147 @@
+
+require "yast2/execute"
+
+module Y2Storage
+  module EncryptionProcesses
+    # Class representing an APQN used for generating a new secure key, see {SecureKey}.
+    #
+    # For more information, see
+    # https://www.ibm.com/support/knowledgecenter/linuxonibm/com.ibm.linux.z.lxdc/lxdc_zkey_reference.html
+    class Apqn
+      # Location of the lszcrypt command
+      LSZCRYPT = "/sbin/lszcrypt".freeze
+      private_constant :LSZCRYPT
+
+      class << self
+        # All APQNs found in the system
+        #
+        # @return [Array<Apqn>]
+        def all
+          create_apqns
+        end
+
+        # All online APQNs found in the system
+        #
+        # @return [Array<Apqn>]
+        def online
+          all.select(&:online?)
+        end
+
+        private
+
+        # Creates APQNs from the data read in the system
+        #
+        # @return [Array<Apqn>]
+        def create_apqns
+          apqns_data.map { |d| new(*d) }
+        end
+
+        # Data of the APQNs found in the system
+        #
+        # For each APQN, it returns its name (card.domain), type, mode, status and requests, see
+        # {#execute_lszcrypt}.
+        #
+        # @return [Array<Array<String>>]
+        def apqns_data
+          data = execute_lszcrypt.split("\n").map(&:split)
+          return [] if data.size == 2
+
+          # remove header
+          data.shift(2)
+
+          # select card.domain entries
+          data.select { |d| d.first.match?(/.*\..*/) }
+        end
+
+        # Executes lszcrypt command
+        #
+        # Example of lszcrypt output:
+        #
+        #   "CARD.DOMAIN TYPE  MODE        STATUS  REQUESTS\n" \
+        #   "----------------------------------------------\n" \
+        #   "01          CEX5C CCA-Coproc  online         1\n" \
+        #   "01.0001     CEX5C CCA-Coproc  online         1\n" \
+        #   "01.0004     CEX5C CCA-Coproc  online         0\n" \
+        #   "01.0005     CEX5C CCA-Coproc  online         0"
+        #
+        # @return [String]
+        def execute_lszcrypt
+          Yast::Execute.locally!(LSZCRYPT, stdout: :capture)
+        rescue Cheetah::ExecutionFailed
+          ""
+        end
+      end
+
+      # Card number
+      #
+      # @return [String] e.g., "01"
+      attr_reader :card
+
+      # Domain number
+      #
+      # @return [String] e.g., "0001"
+      attr_reader :domain
+
+      # Card type
+      #
+      # @return [String] e.g., "CEX5C"
+      attr_reader :type
+
+      # Card mode
+      #
+      # @return [String] e.g., "CCA-Coproc"
+      attr_reader :mode
+
+      # APQN status
+      #
+      # @return [String] e.g., "online", "offline"
+      attr_reader :status
+
+      # Constructor
+      #
+      # @param name [String]
+      # @param type [String]
+      # @param mode [String]
+      # @param status [String]
+      # @param _others [Array<String>]
+      def initialize(name, type, mode, status, *_others)
```

Comment:
You are introducing a `*_others` argument just for convenience, to be able to do `new(*d)`. To be honest, I would rather prefer a more explicit approach and not polluting the arguments list. For instance (there are more ways to do so):

```ruby
name, type, mode, status, _ = d
new(name, type, mode, status)
```

If you still want to have such a convenience constructor, I would write it as a separate one that does the lifting and finally relies on the regular `new`.